pubsub/subscription.go

```diff
@@ -111,6 +115,19 @@ func (subs *SubscriptionIterator) Next() (*Subscription, error) {
 	return &Subscription{c: subs.c, name: subName}, nil
 }
 
+// NextConfig returns the next subscription config. If there are no more subscriptions, iterator.Done will be returned.
```

Comment:
I'm not really sure how the iterator is used elsewhere, but I can see that `it.Next` is captured in a closure for the `next` field. Is there any concern that a user will mix calls of `Next` (for the Subscription) and `NextConfig` (for the SubscriptionConfig) that will pull from the same iterator, thereby "losing" items?

Reply:
I sort of see your point. I'll add a note to `NextConfig` that this is shared with `Next` and make sure to create separate iterator instances if you want to use both.

Reply:
Sounds good. You could also make it return both, like `NextWithConfig() (*Subscription, *SubscriptionConfig, error)`. That doesn't help with using `Next` and "losing" the config part - although that might be OK TBH, with a comment to use the `NextWithConfig` version if they want that.

Reply:
I'm fairly certain that most users probably want to retrieve the config only, and that the existing `Next` feature was probably unused. Retrieving `*Subscription` is only useful if you want to start multiple `Receive` calls, which I don't think is a common use case. I'm tempted to keep this as is for simplicity (retrieving both `Subscription` and `SubscriptionConfig` is too much for users).

Reply:
SGTM!